app/src/main/java/org/mozilla/focus/biometrics/BiometricAuthenticationDialogFragment.kt

```diff
@@ -86,16 +92,35 @@ class BiometricAuthenticationDialogFragment : AppCompatDialogFragment(), Lifecyc
         handler = null
     }
 
+    override fun onStop() {
+        super.onStop()
+        ioScope.cancel()
+    }
+
     override fun onDestroyView() {
         super.onDestroyView()
         _binding = null
     }
 
     private fun biometricNewSessionButtonClicked() {
         requireComponents.tabsUseCases.removePrivateTabs()
+        deleteAllTopSites()
+        requireComponents.appStore.dispatch(AppAction.ShowHomeScreen)
         dismiss()
     }
 
+    private fun deleteAllTopSites() {
```

Comment:
I think we should create a separate integration file were to move this kind of logic, it will not be a big step into separation of concerns, but it will be an improvement

Reply:
done